### DISCUSSION TITLE
Error when deleting sub-file readme

### DISCUSSION THREAD

**Discussion by ThienObito** [community/community#63969] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I wrote an extra file to create a snake game on my profile but I don't know how to delete it and I got my token exposed, how do I remove it from the readme page, I used manual measures to go to that file and press delete path but it still doesn't work , Thanks

**Comment by GhostOf0days** [comment, marked as answer]
If your token is exposed, please contact [Github Support](https://support.github.com), and ask them to remove it, explaining how and what sensitive data was exposed.

**Comment by 0dayhunter** [comment]
If you have accidentally exposed sensitive information like your token in a public repository and need to remove it, and you're also facing issues deleting a file from your repository, here's a step-by-step guide to help you:

**1. Remove Sensitive Information:**

If you've accidentally exposed sensitive information like your token, you should revoke that token immediately and generate a new one. Follow these steps:

1. **Revoke Token:** Go to the settings of your GitHub account, navigate to "Developer settings," and then "Personal access tokens." Find the token you exposed and revoke it.

2. **Generate New Token:** Generate a new personal access token with only the permissions you need. Be sure to keep this token secure and avoid exposing it in your code.

**2. Delete the File:**

If you're facing issues deleting a file from your repository, follow these steps:

1. **Clone the Repository:** Open a terminal or command prompt, navigate to a directory where you want to clone your repository, and run:
   
   ```
   git clone <repository_url>
   ```

2. **Delete the File Locally:** Use your operating system's file manager or terminal to navigate to the cloned repository directory. Delete the file that you want to remove.

3. **Commit and Push:** In your terminal, navigate to the repository directory, and execute the following commands:

   ```
   git add .
   git commit -m "Removed sensitive file"
   git push origin main
   ```

4. **GitHub Web Interface:**
   - Open your repository on GitHub through your web browser.
   - Go to the file you want to delete.
   - Click on the "Trashcan" icon or the "Delete" button to delete the file.
   - Commit the deletion by clicking the "Commit changes" button.

**3. Update README:**

If you've exposed your token in the README file and want to remove it, follow these steps:

1. **Clone the Repository:** If you haven't already, clone your repository to your local machine.

2. **Edit the README Locally:** Use a text editor to open the README file in the cloned repository. Remove any sensitive information, including the exposed token.

3. **Commit and Push:** Save the changes to the README file, then commit and push the changes as mentioned earlier.

4. **GitHub Web Interface:**
   - Open your repository on GitHub through your web browser.
   - Go to the README file.
   - Click the "Edit" (pencil) icon to edit the file.
   - Paste your updated content.
   - Commit the changes by clicking the "Commit changes" button.

Always be cautious with sensitive information and ensure that you follow best practices for securing your code and credentials.
